src/js.cookie.js

```diff
@@ -67,7 +67,7 @@
 					attributes.expires && '; expires=' + attributes.expires.toUTCString(), // use expires attribute, max-age is not supported by IE
 					attributes.path    && '; path=' + attributes.path,
 					attributes.domain  && '; domain=' + attributes.domain,
-					attributes.secure  && '; secure'
+					(attributes.secure  && '; secure') || '' // use '' for falsy values, because [NaN, false, 0].join('') === 'NaNfalse0;
```

Comment:
> In that case we would clearly state that it'd be false usage and needed be fixed by them imo.

Any behavior that is not documented should not be relied upon. What we could do is to explicitly declare which types each option accepts.

In the `domain` case, the only falsy value that could be passed would be `''`, which will evaluate to an empty domain and be ignored upon `.join('')`:

```shell
> ['' && '; domain='].join('')
""
```